SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+			EAutomationExpectedErrorFlags::MatchType::Contains, 1);
+
+		// The two single strategies plus the default strat require 3 vitual workers, but we only have 2.
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(ComponentStrategyConfig, 2));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_2_single_cell_grid_strats_and_default_strat_WHEN_set_virtual_worker_ids_called_with_3_ids_THEN_no_error_is_logged)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+
+		std::vector<StrategyConfig> ComponentStrategyConfig
+		{
+			{UGridBasedLBStrategy::StaticClass(), {ADefaultPawn::StaticClass()}},
+			{UGridBasedLBStrategy::StaticClass(), {ASpecialPawn::StaticClass()}}
+		};
+
+		// The two single strategies plus the default strat require 3 vitual workers.
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(ComponentStrategyConfig, 3));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_default_strat_WHEN_requires_handover_called_THEN_returns_false)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy({}, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckRequiresHandover(this, false));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_single_cell_grid_strat_and_default_strat_WHEN_requires_handover_called_THEN_returns_true)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+
+		std::vector<StrategyConfig> ComponentStrategyConfig
+		{
+			{UGridBasedLBStrategy::StaticClass(), {ADefaultPawn::StaticClass()}}
+		};
+
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(ComponentStrategyConfig, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckRequiresHandover(this, true));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_layered_strat_of_default_strat_WHEN_who_should_have_auth_called_THEN_return_1)
+	{
+		AutomationOpenMap("/Engine/Maps/Entry");
+		ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld());
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy({}, {}));
+		ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategysLocalWorker(1));
+		ADD_LATENT_AUTOMATION_COMMAND(FSpawnDefaultPawnAtLocation(TEXT("Actor1"), FVector(0, 0, 0)));
+		ADD_LATENT_AUTOMATION_COMMAND(FCheckWhoShouldHaveAuthority(this, "Actor1", 1));
+		ADD_LATENT_AUTOMATION_COMMAND(FCleanup());
+
+		return true;
+	}
+
+	LAYEREDLBSTRATEGY_TEST(Given_two_actors_of_same_type_at_same_position_WHEN_who_should_have_auth_called_THEN_return_same_for_both)
+	{
```

Comment:
very minor nit. Group the pawn tests together.

Reply:
done